src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -180,11 +180,20 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
             }
 
             var constructorArguments = analyzedArguments.ConstructorArguments;
-            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.ToImmutableAndFree();
+
+            // If the arguments didn't have a natural type, they're not constants, and so would already have diagnostics from the rest of binding.
+            var ignoredDiagnostics = new DiagnosticBag();
+            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.SelectAsArray(
+                (arg, thisAndDiagnostics) => thisAndDiagnostics.@this.BindToNaturalType(arg, thisAndDiagnostics.ignoredDiagnostics),
+                (@this: this, ignoredDiagnostics));
             ImmutableArray<string> boundConstructorArgumentNamesOpt = constructorArguments.GetNames();
-            ImmutableArray<BoundExpression> boundNamedArguments = analyzedArguments.NamedArguments;
+            ImmutableArray<BoundExpression> boundNamedArguments = analyzedArguments.NamedArguments.SelectAsArray(
```

Comment:
>analyzedArguments.NamedArguments [](start = 66, length = 32)

See if we want to change ```BindAttributeArguments``` to keep it as a builder since we are going to allocate a new array here.